tools/hsmtool.c

```diff
@@ -373,11 +373,12 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
 }
 
 static int dumponchaindescriptors(const char *hsm_secret_path, const char *passwd,
-				  const bool private)
+				  const bool private, const bool testnet)
 {
 	struct secret hsm_secret;
 	u8 bip32_seed[BIP32_ENTROPY_LEN_256];
 	u32 salt = 0;
+	u32 version = testnet ? BIP32_VER_TEST_PRIVATE : BIP32_VER_MAIN_PRIVATE;
```

Comment:
Can we infer this from the directory structure or some other info source?

Reply:
No, as we can use all commands of the `hsmtool` on a `hsm_secret` without depending on its environment at all
and this would be quite convoluted too..

Reply:
We have all of these parameters in `chainparams.c`, so why not use them? We can even steal the `--network` option from `lightningd` itself :-)

Reply:
This'd mean bringing the `opt()` system here: that's a much bigger change :/